src/TropicalGeometry/groebner_basis.jl

```diff
@@ -0,0 +1,171 @@
+###
+# Computing (tropical) Groebner bases in Oscar
+# ============================================
+#
+# For a definition of tropical Groebner basis see Section 2.4 in:
+#   D. Maclagan, B. Sturmfels: Introduction to tropical geometry
+# To see how they can be computed using standard bases see:
+#   T. Markwig, Y. Ren: Computing tropical varieties over fields with valuation
+###
+
+
+
+#=======
+return true if f is homogeneous (w.r.t. total degree)
+return false otherwise
+=======#
+function is_homogeneous(f::Union{AbstractAlgebra.Generic.MPoly{K},fmpq_mpoly,fmpz_mpoly} where {K})
```

Comment:
There is a reason that we don't have `is[_]homogenous` for arbitrary multivariate polynomials. How does introducing this function work together with the commutative algebra design @wdecker? Should this be added or renamed to an internal function?

Reply:
Ping @YueRen

Reply:
I think renaming for now is a good idea; also, for ideals, is_homogeneous is misleading as it only tests the generators. Can we find a better name for ideals? I will provide an is_homogeneous function  in full generality (for all gradings) soon.

Reply:
I renamed it now to `_is_homogeneous`. I also renamed the other `is_homogeneous` for ideals to `has_homogeneous_generators`.